store/rootmulti/store.go

```diff
@@ -74,9 +74,11 @@ type Store struct {
 
 	interBlockCache types.MultiStorePersistentCache
 
-	listeners map[types.StoreKey][]types.WriteListener
+	listeners map[types.StoreKey]*types.MemoryListener
 
 	metrics metrics.StoreMetrics
+
+	listenersMx sync.Mutex
```

Comment:
can we add a godoc what this mutex does? Slightly lost why its on existing methods?

Reply:
added comments.